File: common/src/main/java/dev/obscuria/fragmentum/registry/DeferredEntity.java
package dev.obscuria.fragmentum.registry;

import net.minecraft.core.Holder;
import net.minecraft.world.entity.Entity;
import net.minecraft.world.entity.EntityType;

import java.util.function.Supplier;

public class DeferredEntity<T extends Entity> extends Deferred<EntityType<?>, EntityType<T>>
{
    public DeferredEntity(Supplier<Holder<EntityType<?>>> supplier)
    {
        super(supplier);
    }
}

File: common/src/main/kotlin/dev/obscuria/fragmentum/api/registry/DeferredEntity.kt
package dev.obscuria.fragmentum.api.registry

import net.minecraft.core.Holder
import net.minecraft.world.entity.Entity
import net.minecraft.world.entity.EntityType
import java.util.function.Supplier

class DeferredEntity<T : Entity>(supplier: Supplier<Holder<EntityType<*>>>)
    : Deferred<EntityType<*>, EntityType<T>>(supplier)